test/data/test_dataset.py

```diff
@@ -246,7 +244,7 @@ def test_csv_dataset_quotechar(self):
 
         with tempfile.NamedTemporaryFile(dir=self.test_dir) as f:
             for example in example_data:
-                f.write(six.b("{}\n".format(",".join(example))))
+                f.write("{}\n".format(",".join(example)).encode("latin-1"))
```

Comment:
Where is the need for encode coming from?

Reply:
In particular latin-1 vs. UTF-8?

Reply:
From the six documentation:

> `six.b`_(data)_
>
> A “fake” bytes literal. _data_ should always be a normal string literal. ... In Python 3, _data is_ encoded with the latin-1 encoding to bytes.

https://six.readthedocs.io/#six.b

From the six code:
```python
if PY3:
    def b(s):
        return s.encode("latin-1")
```